src/apb/engine.py

```diff
@@ -291,14 +292,29 @@ def touch(fname, force):
         open(fname, 'a').close()
 
 
-def update_spec(project):
+def update_spec(project, ignore_deps):
     spec = get_spec(project)
     spec_path = os.path.join(project, SPEC_FILE)
+    roles_path = os.path.join(project, ROLES_DIR)
 
     # ID specfile if it hasn't already been done
     if 'id' not in spec:
         gen_spec_id(spec, spec_path)
 
+    if not ignore_deps:
+        expected_deps = load_source_dependencies(roles_path)
+        if 'required' not in spec:
```

Comment:
This block of code is only executed if the ignore_deps flag isn't set. There's no need to go through the templating process if the user skips it. These checks are so the templating process doesn't break.